### ISSUE TITLE
Neural_Search 模块在自构建数据集上，无监督训练 Simcse 时，loss 值始终维持在某个值附近

### ISSUE THREAD

**Issue by Nirvana-2021** [PaddlePaddle/PaddleNLP#1915]
版本\环境：
paddlenlp 2.1.1
paddlepaddle-gpu 2.2.0
Linux
python 3.6

问题：
(1) 在 Paddle 提供的样例数据中，经过训练、测试，Neural_Search 模块功能都是正常的;
_**(2) 但是，将样例数据换成自己构建的数据后，Simcse 的无监督训练过程中，Loss 值始终维持在某个值（约 4.2）附近，自构建数据样例，见附件（train_unsupervised_demo.csv, dev_demo.csv）
[dev_demo.csv](https://github.com/PaddlePaddle/PaddleNLP/files/8449365/dev_demo.csv)
[train_unsupervied_demo.csv](https://github.com/PaddlePaddle/PaddleNLP/files/8449263/train_unsupervied_demo.csv)**_
(3) 若将每篇新闻对应的标题作为 query 添加到自构建的 train_unsupervised.csv, 无监督训练 simcse 的 loss 也能不断降低，最终收敛。

**Comment by w5688414** [comment]
我没看出具体原因是啥，但是您的训练文本过长，我们的预设的最大长度是64，你可以考虑一下分句，使用短句进行训练。

**Comment by Nirvana-2021** [comment]
> 我没看出具体原因是啥，但是您的训练文本过长，我们的预设的最大长度是64，你可以考虑一下分句，使用短句进行训练。

仅仅将每篇新闻的标题（字数大约 30 字）作为 train_unsupervised.csv， 问题还是存在的

**Comment by Nirvana-2021** [comment]
> 我没看出具体原因是啥，但是您的训练文本过长，我们的预设的最大长度是64，你可以考虑一下分句，使用短句进行训练。

仅仅将每篇新闻的标题（字数大约 30 字）作为 train_unsupervised.csv， 还是同样的问题。

**Comment by tianxin1860** [comment]
可以把你用的数据发我们，我们具体看下什么原因。

**Comment by Nirvana-2021** [comment]
> 可以把你用的数据发我们，我们具体看下什么原因。

链接：https://pan.baidu.com/s/1ly_p-xWANryRK8GoRjEFUQ 
提取码：0420

**Comment by Nirvana-2021** [comment]
> 可以把你用的数据发我们，我们具体看下什么原因。

链接：https://pan.baidu.com/s/1ILvjiMi3hnDUj0zqaxSMJw 
提取码：0420

**Comment by tianxin1860** [comment]
好的，我们看下什么问题

**Comment by w5688414** [comment]
@[Nirvana-2021](https://github.com/Nirvana-2021)

下面是我的命令：

```
python -u -m paddle.distributed.launch --gpus '4,5,6,7' \
	train.py \
	--device gpu \
	--save_dir ./checkpoints/ \
	--batch_size 64 \
	--learning_rate 5E-5 \
	--epochs 3 \
	--save_steps 2000 \
	--eval_steps 100 \
	--max_seq_length 64 \
	--dropout 0.2 \
    --output_emb_size 256 \
	--train_set_file "./recall/train_unsupervised.csv" \
	--test_set_file "./recall/test.csv" 
```
日志输出：
```
[2022-04-24 10:21:05,018] [    INFO] - Already cached /root/.paddlenlp/models/ernie-1.0/vocab.txt
global step 10, epoch: 1, batch: 10, loss: 0.36822, speed: 1.91 step/s
global step 20, epoch: 1, batch: 20, loss: 0.08188, speed: 3.17 step/s
global step 30, epoch: 1, batch: 30, loss: 0.04715, speed: 2.96 step/s
global step 40, epoch: 1, batch: 40, loss: 0.03500, speed: 2.92 step/s
global step 50, epoch: 1, batch: 50, loss: 0.04099, speed: 3.09 step/s
global step 60, epoch: 1, batch: 60, loss: 0.04319, speed: 2.91 step/s
global step 70, epoch: 1, batch: 70, loss: 0.02173, speed: 3.07 step/s
global step 80, epoch: 1, batch: 80, loss: 0.01466, speed: 2.91 step/s
global step 90, epoch: 1, batch: 90, loss: 0.01337, speed: 2.99 step/s
global step 100, epoch: 1, batch: 100, loss: 0.02927, speed: 3.19 step/s
global step 110, epoch: 1, batch: 110, loss: 0.02364, speed: 2.80 step/s
global step 120, epoch: 1, batch: 120, loss: 0.03533, speed: 2.92 step/s

......
```

数据集只用了title：

```
外媒：特朗普6月将赴法参加诺曼底登陆75周年纪念活动
西媒盘点：历史不该遗忘的七位女数学家
阿桑奇七年使馆生活细节曝光：经常用手吃饭很少洗澡
外媒：苏丹国防部长宣布已将总统巴希尔“关押在安全的地方”
西媒称维基揭秘网是对爆料者的背叛：在美俄两大国间挑唆战争
外媒：韩国总统文在寅再“出马”欲拉特朗普重回美朝核谈判
阿尔及利亚民众继续示威要求变革
日本餐饮零售业开始全面禁烟：让带小孩的客人放心用餐
“你不该坐在这儿”美副总统在安理会公然威胁委内瑞拉大使
阿桑奇被捕时，喊出的竟是这个人的名字——
......
```

**Comment by Nirvana-2021** [comment]
更新代码至最新版本，在相同的 10 万条数据上，loss 值变化如下：
global step 10, epoch: 1, batch: 10, loss: 4.23565, speed: 1.33 step/s
global step 20, epoch: 1, batch: 20, loss: 4.19754, speed: 1.67 step/s
global step 30, epoch: 1, batch: 30, loss: 4.21234, speed: 1.65 step/s
global step 40, epoch: 1, batch: 40, loss: 4.17117, speed: 1.57 step/s
global step 50, epoch: 1, batch: 50, loss: 4.22895, speed: 1.70 step/s
global step 60, epoch: 1, batch: 60, loss: 4.20006, speed: 1.56 step/s
global step 70, epoch: 1, batch: 70, loss: 4.17178, speed: 1.54 step/s
global step 80, epoch: 1, batch: 80, loss: 4.20667, speed: 1.56 step/s
global step 90, epoch: 1, batch: 90, loss: 4.24917, speed: 1.46 step/s
global step 100, epoch: 1, batch: 100, loss: 4.19574, speed: 1.61 step/s
global step 110, epoch: 1, batch: 110, loss: 4.20496, speed: 1.63 step/s
global step 120, epoch: 1, batch: 120, loss: 4.20039, speed: 1.62 step/s
global step 130, epoch: 1, batch: 130, loss: 4.15629, speed: 1.50 step/s
global step 140, epoch: 1, batch: 140, loss: 4.21160, speed: 1.55 step/s
global step 150, epoch: 1, batch: 150, loss: 4.18189, speed: 1.52 step/s
global step 160, epoch: 1, batch: 160, loss: 4.20814, speed: 1.38 step/s
global step 170, epoch: 1, batch: 170, loss: 4.20895, speed: 1.54 step/s
global step 180, epoch: 1, batch: 180, loss: 4.19990, speed: 1.51 step/s
global step 190, epoch: 1, batch: 190, loss: 4.23599, speed: 1.42 step/s
global step 200, epoch: 1, batch: 200, loss: 4.19934, speed: 1.38 step/s
global step 210, epoch: 1, batch: 210, loss: 4.20561, speed: 1.50 step/s
global step 220, epoch: 1, batch: 220, loss: 4.16574, speed: 1.50 step/s
global step 230, epoch: 1, batch: 230, loss: 4.17004, speed: 1.54 step/s
global step 240, epoch: 1, batch: 240, loss: 4.20078, speed: 1.52 step/s
global step 250, epoch: 1, batch: 250, loss: 4.18563, speed: 1.36 step/s
global step 260, epoch: 1, batch: 260, loss: 4.17251, speed: 1.49 step/s
global step 270, epoch: 1, batch: 270, loss: 4.21160, speed: 1.43 step/s
global step 280, epoch: 1, batch: 280, loss: 4.19814, speed: 1.56 step/s
global step 290, epoch: 1, batch: 290, loss: 4.19875, speed: 1.46 step/s
global step 300, epoch: 1, batch: 300, loss: 4.17259, speed: 1.51 step/s
global step 310, epoch: 1, batch: 310, loss: 4.19149, speed: 1.53 step/s
global step 320, epoch: 1, batch: 320, loss: 4.20303, speed: 1.40 step/s
global step 330, epoch: 1, batch: 330, loss: 4.21734, speed: 1.57 step/s
global step 340, epoch: 1, batch: 340, loss: 4.18296, speed: 1.49 step/s
global step 350, epoch: 1, batch: 350, loss: 4.15554, speed: 1.48 step/s
global step 360, epoch: 1, batch: 360, loss: 4.21446, speed: 1.61 step/s
global step 370, epoch: 1, batch: 370, loss: 4.17501, speed: 1.56 step/s
global step 380, epoch: 1, batch: 380, loss: 4.19517, speed: 1.56 step/s
global step 390, epoch: 1, batch: 390, loss: 4.19176, speed: 1.46 step/s
global step 400, epoch: 1, batch: 400, loss: 4.17986, speed: 1.58 step/s
global step 410, epoch: 1, batch: 410, loss: 4.19225, speed: 1.55 step/s
global step 420, epoch: 1, batch: 420, loss: 4.19567, speed: 1.44 step/s
global step 430, epoch: 1, batch: 430, loss: 4.15250, speed: 1.54 step/s

**Comment by Nirvana-2021** [comment]
两个问题确认一下：
1. 同样一批数据，为啥大佬们的 loss 的值那么小？
2. 将 10 万数据划分成训练集、测试集、验证集、语料库时，除运行 process.py 外，是否有其他操作？

**Comment by Nirvana-2021** [comment]
再补充一个问题：
3. simcse 运行的依赖环境和第三方库是什么？

**Comment by w5688414** [comment]
[train_unsupervised.csv](https://github.com/PaddlePaddle/PaddleNLP/files/8617698/train_unsupervised.csv)

**Comment by w5688414** [comment]
```
paddlepaddle-gpu                   2.2.2
paddlenlp                          2.2.6
```
@Nirvana-2021

**Comment by Nirvana-2021** [comment]
重新搭建运行环境，更新底层依赖项，问题已解决，版本参考见附件：
[requirements.total.txt](https://github.com/PaddlePaddle/PaddleNLP/files/8683698/requirements.total.txt)
